pages/common/fgrep.md

```diff
@@ -0,0 +1,28 @@
+# fgrep
+
+> Matches patterns in files.
+> Supports simple patterns and regular expressions.
+
+- Search for an exact string in a file:
+
+`fgrep {{search_string}} {{path/to/file}}`
+
+- Search only lines that match entirely in files:
+
+`fgrep -x {{path/to/file1}} {{path/to/file2}}`
+
+- Count the number of lines that match the given string in a file:
+
+`fgrep -c {{search_string}} {{path/to/file}}`
+
+- Show line number of file with the line matched:
+
+`fgrep -n {{search_string}} {{path/to/file}}`
+
+- Display all lines except those that contain the regular expression:
+
+`fgrep -v {{^regex$}} {{path/to/file}}`
+
+- Search the file names that match the regular expression:
```

Comment:
I think this might be a bit misleading. Here's an extract from the man page:

```
       -l, --files-with-matches
              Suppress normal output; instead print the  name  of  each  input
              file  from  which  output would normally have been printed.  The
              scanning will stop on the first match.
```

In your explanation, it sounds like it searches the filenames themselves, instead of their content. Speaking of, you include a regex there. I thought you needed to include the `--perl-regexp` (`-P`) flag to activate this feature?

Anyway, how about we reword this - perhaps this will read better?
```suggestion
- Display filenames whose content matchs the regular expression at least once:
```